0036-spsp-pull-payments/0036-spsp-pull-payments.md

```diff
@@ -0,0 +1,135 @@
+---
+title: SPSP Pull Payments
+draft: 1
+---
+# SPSP Pull Payments
+
+## Preface
+
+This document describes how to conduct pull payments via [STREAM](../0029-stream/0029-stream.md) once the payment details have been exchanged via the [Simple Payment Setup Protocol](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md) (SPSP).
+
+## Introduction
+
+### Motivation
+
+The Simple Payment Setup Protocol (SPSP) only describes how the details required for a STREAM connection are exchanged and how a simple push payment is made. However, it lacks a detailed explaination of how a pull payment is conducted using this connection. 
+
+### Scope
+
+This document specifies how to conduct a pull payment once a STREAM connection has been set up. It is intended for use by end-user applications.
+
+### Operation Overview
+
+Any SPSP server will expose an HTTPS endpoint called the SPSP Endpoint. The client can query this endpoint to get information about about the server's connection details, namely ILP address and a shared secret. The endpoint MAY also contain additional information for displaying purposes. The client can set up a STREAM connection and pull multiple ILP payments using the details provided by the server. 
+
+## Model of Operation
+
+### Creating a token
+
+A pull payment token SHOULD be be opaque. For the purpose of offline generation, it MAY include negotiated pull payment agreement parameters that SHOULD become obsolete once the SPSP endpoint corresponding to that token has been created. The SPSP server SHOULD store the negotiated pull payment agreement parameters.
```

Comment:
This is a difficult paragraph to parse. There is a lot there that I think needs some explanation.

What is a token?
What are "negotiated pull payment agreement parameters"?

Reply:
I agree, I should define what a _token_ is. There is a brief description of what the _negotiated pull payment agreement parameters_ are in the paragraph following this one but yes, there could be more information on that, too.

I'm wondering, however, how detailed this section should be. The way I understood our previous discussions is that there should not be too much detail about the negotiation phase. If anything, that should become a different spec.

Reply:
> there should not be too much detail about the negotiation phase. If anything, that should become a different spec.

That's true but there is a lot of new language in here without context. Some light background is needed, especially while the other spec doesn't exist and can't be referenced.